### PR TITLE
Fix (Adminhtml-Customer-Edit): allowed countries of selected website aren't considered when editing existing addresses

### PR DESCRIPTION
**Pull request by DeineTuer-GmbH**
### Description (*)

This PR enables the editing of customers with addresses using countries that aren't allowed in the default config (`General > Countries Options > Allow Countries [general/country/allow]`) but are in `Website` scope, without the need to change the country first.

This is achieved by taking the customer's store id into account when determining the allowed countries to fill the select options. Previously the input was always filled with the allowed country of the default store (admin, store id 0). Setting the customer's store id in select and multiselect attributes is necessary because the way in which values are set in `Mage_Adminhtml_Block_Widget_Form::_setFieldset()` line 208 and 210 respectively.

### Related Pull Requests

* None

### Fixed Issues (if relevant)

* #1762 

### Manual testing scenarios (*)

1. Set system config `General > Countries Options > Allow Countries [general/country/allow]` so that at least one country is allowed in `Website` scope that is not allowed in default config scope. 
2. Have a customer for this website, log into it and create an address that uses a country that is only allowed in `Website` scope.
3. Navigate to Adminhtml Customer/Edit, edit the customer and select addresses.
4. See the created address with the country selected in the country input select.

### Questions or comments

This PR can't fix cases in which one decides to set the allowed countries in `Store View` scope because a customer is assigned to a website (and within the database to a store id it was created in). This is true in some circumstances and additionally depends on the configured store scenario (how many websites/stores/store views and in what distribution).

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ADDISON74** [comment]
I think this PR should be better explained. Magento allows in the Backend to select the countries in which the store is available (General/Countries Options). If there is only one store you will not find that there is an issue because both in Frontend in editing the customer's account and in Backend in editing the customer's in the drop-down list there will be only the allowed countries set in the configuration.

The situation changes when you have several stores and each store has its own countries selection. Magento will show you only those in the general configuration, not those specific to the store. That's why it is necessary to use the store ID in order to get the correct selection of countries.

**Comment by deinetuer** [comment]
@ADDISON74 Thanks for the feedback! We overhauled the explanation of the PR and created an accompaning issue (#1762) to explain the problem even further.

**Review by Sdfendor — APPROVED** [review]

**Review by tobihille — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Addresses.php [hunk 1]
```diff
@@ -127,10 +127,25 @@ public function initForm()
             Mage::helper('adminhtml/addresses')
                 ->processStreetAttribute($attributes['street']);
         }
+
+        $customerStoreId = null;
+        if ($customer->getId()) {
+            $customerStoreId = Mage::app()->getWebsite($customer->getWebsiteId())->getDefaultStore()->getId();
+        }
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Addresses.php:134]: I see that the code block is moved from line 163. But if a website has more than one store, with mutually exclusive country list, e.g., the website's default store is store1, which has only France, store2 has only Germany, etc. Wouldn't non-French customers get incorrect country list? Is this case tested?

```suggestion
        $customerStoreId = $customer->getStoreId(); // null if new customer
```

Not tested, but I think the above should work for `suffix` and `prefix` later on, lines 180 and 194.

File: app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Addresses.php [hunk 2]
```diff
@@ -127,10 +127,25 @@ public function initForm()
...
             $attribute->unsIsVisible();
+            
+            /* Sets store id to attribute if select or multiselect because values are loaded store specific.
+               If not set values of default store are loaded which can be undesirable (e.g. country list in customer
+               addresses). */
+            /** @see Mage_Adminhtml_Block_Widget_Form::_setFieldset() lines 208 & 210 */
+            $frontendInputType = $attribute->getFrontend()->getInputType();
+            if ($customerStoreId !== null && ($frontendInputType === 'select' || $frontendInputType === 'multiselect')) {
+                $attribute->setStoreId($customerStoreId);
+            }
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Addresses.php:148]:
```suggestion
            $attribute->setStoreId($customerStoreId); // For country list
```
I think it's unnecessary to check for `null` and input type.

**Review by kiatng — COMMENTED** [review]
Suggestions are note tested.

**Comment by fballiano** [comment]
@DeineTuer-GmbH can you rebase to `main` branch and check the comments to the code?

**Comment by fballiano** [comment]
This PR could hide a problem because of https://github.com/OpenMage/magento-lts/pull/1754#discussion_r950628781 and since it's now on a wrong branch and with no feedback since many month, sorry but I'll close it.

Hope it will be reopened by the original author with answers to the feedbacks :-)

**Pull request closed without merging**